Update django to version 4.2 and fix tests

ttys0dev opened the pull request.

mlissner commented:
Is this ready for review? We have a practice here of requesting a review (usually mine) once something is ready.

If so, can you explain your process for this upgrade and some of the changes that you made? I usually do the django upgrades one version at a time to make sure we catch and fix everything, so it makes me a bit nervous to do a huge upgrade in one fell swoop like this (though it'd sure be nice to say it was done!)

mlissner commented:
Huh. Django 4.2 drops support from Postgresql 11, which I think we still use in prod. We'll have to upgrade that before we can merge this.

ttys0dev commented:
> Is this ready for review?

Yeah, should be ready.

> If so, can you explain your process for this upgrade and some of the changes that you made? I usually do the django upgrades one version at a time to make sure we catch and fix everything, so it makes me a bit nervous to do a huge upgrade in one fell swoop like this (though it'd sure be nice to say it was done!)

Updated the django version then ran migrations following [the instructions](https://github.com/freelawproject/courtlistener/blob/main/MIGRATIONS.md) and fixed whatever errors were getting hit by the tests still(which seemed to be mostly stuff which needed primary key existence checks).

ttys0dev commented:
> Huh. Django 4.2 drops support from Postgresql 11, which I think we still use in prod. We'll have to upgrade that before we can merge this.

Yeah, 12 is the minimum supported version now, what version of postgres is the production setup using?

quevon24 commented:
index_together option is deprecated in Django 4.2, you need to replace it with "indexes" option in the models: https://docs.djangoproject.com/en/4.2/releases/4.2/#index-together-option-is-deprecated-in-favor-of-indexes

ttys0dev commented:
> index_together option is deprecated in Django 4.2, you need to replace it with "indexes" option in the models:

Replaced those now.

> That'd be fine if recreating indexes was not a huge deal, except in our case it is because we have huge indexes. Something that makes this worse is that Django also doesn't create indexes `CONCURRENTLY`, so, by default, I think it locks tables while indexes are built.

I think it should be ok since it looks to just be an index rename.

ttys0dev commented:
rebased
